app/main.py

```diff
@@ -35,22 +34,22 @@ def __mul__(self, times: int) -> Distance:
     def __truediv__(self, divisor: int) -> Distance:
         return Distance(round(self.km / divisor, 2))
 
-    def __lt__(self, other: Distance) -> int | float:
-        other_km = other.km if isinstance(other, Distance) else other
+    def __lt__(self, other: Any) -> bool:
+        other_km = self.check_distance(other)
         return self.km < other_km
 
-    def __gt__(self, other: Distance) -> int | float:
-        other_km = other.km if isinstance(other, Distance) else other
+    def __gt__(self, other: Any) -> bool:
+        other_km = self.check_distance(other)
         return self.km > other_km
 
-    def __eq__(self, other: Distance) -> int | float:
-        other_km = other.km if isinstance(other, Distance) else other
+    def __eq__(self, other: Any) -> bool:
+        other_km = self.check_distance(other)
         return self.km == other_km
 
-    def __le__(self, other: Distance) -> int | float:
-        other_km = other.km if isinstance(other, Distance) else other
+    def __le__(self, other: Any) -> bool:
+        other_km = self.check_distance(other)
         return self.km <= other_km
```

Comment:
You ignored comment about total_ordering or returning `not self.function()`, fix that